internal/service/console.go

```diff
@@ -2,14 +2,132 @@ package service
 
 import (
 	"context"
+	"net/http"
+	"sync"
+	"time"
 
 	api "github.com/flightctl/flightctl/api/v1alpha1"
 	"github.com/flightctl/flightctl/internal/api/server"
 	"github.com/flightctl/flightctl/internal/flterrors"
 	"github.com/flightctl/flightctl/internal/store"
+	"github.com/go-chi/chi/v5"
 	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
 )
 
+var upgrader = websocket.Upgrader{
+	CheckOrigin: func(r *http.Request) bool {
+		return true // Allow connections from any origin
+	},
+}
+
+func (h *WebsocketHandler) HandleDeviceConsole(w http.ResponseWriter, r *http.Request) {
+	orgId := store.NullOrgId
+	deviceName := chi.URLParam(r, "name")
+
+	h.log.Infof("websocket console connection requested for device: %s", deviceName)
+	consoleSession, err := h.consoleSessionManager.StartSession(r.Context(), orgId, deviceName)
+	// check for errors
+	if err != nil {
+		switch err {
+		case flterrors.ErrResourceNotFound:
+			h.log.Errorf("console requested for unknown device: %s", deviceName)
+			http.Error(w, "Device not found", http.StatusNotFound)
+		default:
+			h.log.Errorf("There was an error retrieving DB from database during console request: %v", err)
+			http.Error(w, "Internal server error", http.StatusInternalServerError)
+		}
+		return
+	}
+
+	// Upgrade the HTTP connection to a WebSocket
```

Comment:
should we validate ws related headers exist before upgrade?

Reply:
I haven't seen examples of that , at this point we are under the /ws/v1/  API path.

Ahh, investigating a bit more, this is handled by the upgrade function already:

![image](https://github.com/user-attachments/assets/054bb16e-ab38-415b-bf7f-7bee5ef49f28)

Reply:
nice thx for digging